genienlp/util.py

```diff
@@ -599,9 +599,16 @@ def make_data_loader(dataset, numericalizer, batch_size, device=None, train=Fals
         f'answer lengths (min, mean, max): {np.min(answer_lengths)}, {int(np.mean(answer_lengths))}, {np.max(answer_lengths)}'
     )
 
-    if dataset.batch_size_fn == input_tokens_fn:
+    if train:
```

Comment:
Can you explain the design here? Is the different sorting something you need for e2e dialogues?

Reply:
train and eval sets may need different sorting or batching. for example in e2e dialogues, eval set should not be sorted and should be batched based on number of examples instead of tokens.